### PR TITLE
Fix OutputData::input_signing_data

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

**Review by Alex6323 — DISMISSED** [review]

File: sdk/src/wallet/types/mod.rs [hunk 1]
```diff
@@ -66,20 +66,18 @@ impl OutputData {
...
-        } else if required_address.is_ed25519() {
-            if wallet_data.address.inner() == &required_address {
-                // TODO #1279: do we need a check to make sure that `wallet_data.address` and `wallet_data.bip_path` are
-                // never conflicting?
+        // TODO #1279: do we need a check to make sure that `wallet_data.address` and `wallet_data.bip_path` are never
+        // conflicting?
+
+        let chain = self.chain.or_else(|| {
+            if &required_address == wallet_data.address.inner()
+                || required_address.backing_ed25519() == wallet_data.address.inner().as_ed25519_opt()
```
Comment (Alex6323) [inline sdk/src/wallet/types/mod.rs]: edge case: what if both or `None`? :thinking:

**Review by kwek20 — COMMENTED** [review]
we used to return Ok(None);, now no more. do we really need to send Result<Option<>> still?

File: sdk/src/wallet/types/mod.rs (hunk 1 above)
Comment (Thoralf-M) [reply sdk/src/wallet/types/mod.rs]: `wallet_data.address.inner().as_ed25519_opt()` can not be None at this point?

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/src/wallet/types/mod.rs (hunk 1 above)
Comment (Alex6323) [reply sdk/src/wallet/types/mod.rs]: chain implies ed25519, right? So can we remove the first part of the if?

Also to make extra sure we match `Some`s we could then just do:
`if matches!((required_address.backing_ed25519(), wallet_data.address.inner().as_ed25519_opt()), (Some(a), Some(b)) if a == b) { ... }`

Comment (thibault-martinez) [reply sdk/src/wallet/types/mod.rs]: Honestly I don't like these `matches!` with an included `if`, I find them extremely hard to read

Comment (Alex6323) [reply sdk/src/wallet/types/mod.rs]: you can't do that unfortunately :shrug: :

```rust
if let (Some(a), Some(b)) = (required_address.backing_ed25519(), wallet_data.address.inner().as_ed25519_opt()) && a == b { ... }
```

still ... the first part of the `if` isn't necessary, right?

Comment (thibault-martinez) [reply sdk/src/wallet/types/mod.rs]: What first part? `required_address == wallet_data.address.inner()`? It's so we also check address that are not directly Ed25519

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**